test/services/TaiServiceSpec.scala

```diff
@@ -16,78 +16,193 @@
+        |   "data" : [ {
+        |      "componentType" : "EmployerProvidedServices",
+        |      "employmentId" : 12,
+        |      "amount" : 12321,
+        |      "description" : "Some Description",
+        |      "iabdCategory" : "Benefit"
+        |   }, {
+        |      "componentType" : "PersonalPensionPayments",
+        |      "employmentId" : 31,
+        |      "amount" : 12345,
+        |      "description" : "Some Description Some",
+        |      "iabdCategory" : "Allowance"
+        |   } ],
+        |   "links" : [ ]
+        |}""".stripMargin)
 
     val anException = new RuntimeException("Any")
+    lazy val (metrics, timer) = {
 
-    lazy val (service, metrics, timer) = {
+      val timer = MockitoSugar.mock[Timer.Context]
 
-      val fakeSimpleHttp = {
-        if(simulateTaiServiceIsDown) new FakeSimpleHttp(Right(anException))
-        else new FakeSimpleHttp(Left(httpResponse))
-      }
+      val metrics = MockitoSugar.mock[MetricsOperator]
+      when(metrics.startTimer(any())) thenReturn timer
 
-      val timer = MockitoSugar.mock[Timer.Context]
+      (metrics, timer)
+    }
 
-      lazy val taiService: TaiService = new TaiService(injected[Environment], injected[Configuration], fakeSimpleHttp, MockitoSugar.mock[Metrics]) {
+    lazy val fakeSimpleHttp = {
+      if (simulateTaiServiceIsDown) new FakeSimpleHttp(Right(anException))
+      else new FakeSimpleHttp(Left(httpResponse))
+    }
 
-        override val metricsOperator: MetricsOperator = MockitoSugar.mock[MetricsOperator]
-        when(metricsOperator.startTimer(any())) thenReturn timer
-      }
+    def service(simpleHttp: SimpleHttp = fakeSimpleHttp) = {
+
+      val m = metrics
 
-      (taiService, taiService.metricsOperator, timer)
+
+      new TaiService(
+        injected[Environment], injected[Configuration], injected[CircuitBreaker], simpleHttp, MockitoSugar.mock[Metrics]
+      ) {
+        override val metricsOperator: MetricsOperator = m
+      }
     }
   }
 
+
   "Calling TaiService.taxSummary" should {
 
     trait LocalSetup extends SpecSetup {
-      val metricId = "get-tax-components"
+      val metricId: Metric = "get-tax-components"
+    }
+
+    "circuitBreaker must call tai" in new LocalSetup {
+      override lazy val simulateTaiServiceIsDown = false
+      val mockSimpleHttp = MockitoSugar.mock[SimpleHttp]
+      override lazy val httpResponse = HttpResponse(SEE_OTHER)
+
+      when(mockSimpleHttp.get[TaxComponentsResponse](Matchers.any())(Matchers.any(), Matchers.any())(Matchers.any()))
+        .thenReturn(
+          Future.successful(TaxComponentsUnexpectedResponse(httpResponse)),
+          Future.successful(TaxComponentsSuccessResponse(TaxComponents(Seq("EmployerProvidedServices", "PersonalPensionPayments"))))
+        )
+
+      val taiService = service(mockSimpleHttp)
+
+      val result1 = taiService.taxComponents(Fixtures.fakeNino, 2014)
+      await(result1) shouldBe TaiNotAvailable
+      Thread.sleep(500)
```

Comment:
Use `.futureValue` instead of `await` and this will mean you do not need to sleep the thread. You will need to mixin `ScalaFutures`

Same for the other sleeps